esphome fails to compile with 4.2.0 release due to conflicting decls of cyw43_hal_generate_laa_mac

When including both `WiFi.h` and `cyw43.h`, I get this compile error:
```
~/.platformio/packages/framework-arduinopico/libraries/lwIP_Ethernet/src/LwipIntfDev.h:57:17: error: 'void cyw43_hal_generate_laa_mac(int, uint8_t*)' was declared 'extern' and later 'static' [-fpermissive]
```

`cyw43_hal_generate_laa_mac` is defined here as `static`:
https://github.com/earlephilhower/arduino-pico/blob/db4b167aaafc33d235d2b1f1949fcae8d107c10e/libraries/lwIP_Ethernet/src/LwipIntfDev.h#L57

While it is declared here as non-static:
https://github.com/raspberrypi/pico-sdk/blob/efe2103f9b28458a1615ff096054479743ade236/src/rp2_common/pico_cyw43_driver/include/cyw43_configport.h#L125

Removing the static from `LwipIntfDev.h` will fail with multiple definitions during linking, which can be solved by adding `__attribute__((weak))` to the definition in `LwipIntfDev.h`.

--- a/libraries/lwIP_Ethernet/src/LwipIntfDev.h
+++ b/libraries/lwIP_Ethernet/src/LwipIntfDev.h
@@ -53,15 +53,13 @@
 
 // Dup'd to avoid CYW43 dependency
 // Generate a mac address if one is not set in otp
-extern "C" {
-    static void cyw43_hal_generate_laa_mac(__unused int idx, uint8_t buf[6]) {
-        pico_unique_board_id_t board_id;
-        pico_get_unique_board_id(&board_id);
-        memcpy(buf, &board_id.id[2], 6);
-        buf[0] &= (uint8_t)~0x1; // unicast
-        buf[0] |= 0x2; // locally administered
-    }
-};
+static void _cyw43_hal_generate_laa_mac(__unused int idx, uint8_t buf[6]) {
+    pico_unique_board_id_t board_id;
+    pico_get_unique_board_id(&board_id);
+    memcpy(buf, &board_id.id[2], 6);
+    buf[0] &= (uint8_t)~0x1; // unicast
+    buf[0] |= 0x2; // locally administered
+}
 
 
 
@@ -70,8 +68,6 @@ enum EthernetLinkStatus {
     LinkON,
     LinkOFF
 };
-
-extern "C" void cyw43_hal_generate_laa_mac(__unused int idx, uint8_t buf[6]);
 
 template<class RawDev>
 class LwipIntfDev: public LwipIntf, public RawDev {
@@ -356,7 +352,7 @@ bool LwipIntfDev<RawDev>::begin(const uint8_t* macAddress, const uint16_t mtu) {
 #if 1
         // forge a new mac-address from the esp's wifi sta one
         // I understand this is cheating with an official mac-address
-        cyw43_hal_generate_laa_mac(0, _macAddress);
+        _cyw43_hal_generate_laa_mac(0, _macAddress);
 #else
         // https://serverfault.com/questions/40712/what-range-of-mac-addresses-can-i-safely-use-for-my-virtual-machines
         memset(_macAddress, 0, 6);

